lib/active_merchant/billing/gateways/paymentez.rb

```diff
@@ -0,0 +1,273 @@
+        post[:session_id] = options[:session_id] if options[:session_id]
+
+        post[:order] ||= {}
+        post[:order][:amount] = amount(money).to_f
+        post[:order][:vat] = options[:vat] if options[:vat]
+        post[:order][:dev_reference] = options[:dev_reference] if options[:dev_reference]
+        post[:order][:description] = options[:description] if options[:description]
+        post[:order][:discount] = options[:discount] if options[:discount]
+        post[:order][:installments] = options[:installments] if options[:installments]
+        post[:order][:installments_type] = options[:installments_type] if options[:installments_type]
+        post[:order][:taxable_amount] = options[:taxable_amount] if options[:taxable_amount]
+      end
+
+      def add_payment(post, payment)
+        post[:card] ||= {}
+        post[:card][:number] = payment.number
+        post[:card][:holder_name] = payment.name
+        post[:card][:expiry_month] = payment.month
+        post[:card][:expiry_year] = payment.year
+        post[:card][:cvc] = payment.verification_value
+        post[:card][:type] = CARD_MAPPING[payment.brand]
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def commit_raw(object, action, parameters)
+        url = "#{(test? ? test_url : live_url)}#{object}/#{action}"
+
+        begin
+          raw_response = ssl_post(url, post_data(parameters), headers)
+        rescue ResponseError => e
+          raw_response = e.response.body
+        end
+        parse(raw_response)
+      end
+
+      def commit_transaction(action, parameters)
+        response = commit_raw('transaction', action, parameters)
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      def commit_card(action, parameters)
+        response = commit_raw('card', action, parameters)
+        Response.new(
+          card_success_from(response),
+          card_message_from(response),
+          response,
+          authorization: card_authorization_from(response),
+          test: test?,
+          error_code: card_error_code_from(response)
+        )
+      end
+
+      def headers
+        {
+          'Auth-Token' => authentication_code,
+          'Content-Type' => 'application/json'
+        }
+      end
+
+      def success_from(response)
+        !response.include?('error') && (response['status'] || response['transaction']['status']) == 'success'
+      end
+
+      def card_success_from(response)
+        return false if response.include?('error')
+        return true if response['message'] == 'card deleted'
+        response['card']['status'] == 'valid'
+      end
+
+      def message_from(response)
+        if success_from(response)
+          response['transaction'] && response['transaction']['message']
```

Comment:
Wait, does this not just return a bool?

Reply:
Ah it returns the value for the last argument. Neat.

Reply:
Yeah, that's why Ruby 2.3 named its safe navigation operator `&.`, rather than e.g. the more common `?.`: naming it `&.` echoes this pattern.